src/types/tuple.rs

```diff
@@ -124,9 +132,13 @@ impl<'a> Iterator for PyTupleIterator<'a> {
     #[inline]
     fn next(&mut self) -> Option<&'a PyAny> {
         if self.index < self.length {
-            let item = self.tuple.get_item(self.index);
-            self.index += 1;
-            Some(item)
+            match self.tuple.get_item(self.index) {
+                Ok(item) => {
+                    self.index += 1;
+                    Some(item)
+                }
+                _ => None,
+            }
```

Comment:
I think it's fine to use `.expect` here.


```suggestion
            let item = self.tuple.get_item(self.index).expect("index is less than length");;
            self.index += 1;
            Some(item)
```

If you don't feel comfortable about the possble panic, it would also be possible to implement a `#[cfg(not(Py_LIMITED_API))]` version of `PyTupleIterator` which uses `as_slice`.

Reply:
Overall I'm against having safe code that can panic (even if it's induced by very odd human error)
I'll check the second suggestion.